library/kani/src/shadow.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+extern "C" {
+    pub fn __KANI_pointer_object(ptr: *const u8) -> usize;
+    pub fn __KANI_pointer_offset(ptr: *const u8) -> usize;
+}
+
+const MAX_NUM_OBJECTS: usize = 1024;
+const MAX_OBJECT_SIZE: usize = 64;
+
+pub struct ShadowMem {
+    is_init: [[bool; MAX_OBJECT_SIZE]; MAX_NUM_OBJECTS],
+}
+
+impl ShadowMem {
+    pub const fn new() -> Self {
```

Comment:
can you please add `#[unstable]` attribute to the public functions?

Reply:
Makes sense.

Reply:
Done.